lib/collectinfo_analyzer/get_controller.py

```diff
@@ -130,28 +130,16 @@ def _filter_ts_host_dc_ns_dict(
         d: TimestampDict[NodeDict[DatacenterDict[NamespaceDict[dict[str, str]]]]],
         dcs_filter: list[str] | None = None,
         namespaces_filter: list[str] | None = None,
-        principal=False,
     ) -> TimestampDict[NodeDict[DatacenterDict[NamespaceDict[dict[str, str]]]]]:
         """
         Removes dcs and namespaces not found in dcs_filter and namespaces_filter. Edits
         d in place and returns d.
         """
-        if not dcs_filter and not namespaces_filter and not principal:
+        if not dcs_filter and not namespaces_filter:
             return d
 
         for ts, nodes_vals in d.items():
-            principal_ip = None
-
-            if principal:
-                node_id_to_ip = self.log_handler.get_node_id_to_ip_mapping(ts)
-                principal_id = self.log_handler.get_principal(ts)
-                principal_ip = node_id_to_ip[principal_id]
-
             for node_ip, dc_vals in list(nodes_vals.items()):
-                if principal_ip and principal_ip != node_ip:
-                    del nodes_vals[node_ip]
-                    continue
-
```

Comment:
Here you can see we are filtering for xdr info and the principal node but now we will allow a lower-level object to handle the principal filtering for us. This more closely matches the "Cluster" object behavior in live_cluster mode